.bazelignore

```diff
@@ -1,4 +1,38 @@
-e2e/
+e2e/bzlmod
+e2e/bzlmod_bazel_lib_2
+e2e/git_dep_metadata
+e2e/gyp_no_install_script
+e2e/js_image_docker
+e2e/js_image_oci
+e2e/js_run_devserver
+e2e/npm_link_package
+e2e/npm_link_package-esm
+e2e/npm_translate_lock
+e2e/npm_translate_lock_auth
+e2e/npm_translate_lock_empty
+e2e/npm_translate_lock_git+ssh
+e2e/npm_translate_lock_multi
+e2e/npm_translate_lock_partial_clone
+e2e/npm_translate_lock_subdir_patch
+e2e/npm_translate_package_lock
+e2e/npm_translate_yarn_lock
+e2e/package_json_module
+e2e/patch_from_repo
+e2e/pnpm_repo_install
+e2e/pnpm_workspace
+e2e/pnpm_workspace_bazel_lib_2
+e2e/pnpm_workspace_deps
+e2e/pnpm_workspace_rerooted
+e2e/rules_foo
+e2e/update_pnpm_lock
+e2e/update_pnpm_lock_with_import
+e2e/vendored_node
+e2e/vendored_tarfile
+e2e/verify_patches
+e2e/webpack_devserver
+e2e/webpack_devserver_esm
+e2e/worker
+e2e/workspace
```

Comment:
We can no longer just ignore `e2e` because the `write_source_files` target which writes the duplicate e2e with bazel-lib 2.x will puke on the analysis phase test where it globs to check for existence.

However I added some additional tests to ensure all e2es are added here.